// @angular-cool/social-login-buttons/esm2015/lib/facebook-button/cool-facebook-button.component.js
import { ChangeDetectionStrategy, Component, EventEmitter, Input, Output, ViewEncapsulation } from '@angular/core';
export class CoolFacebookButtonComponent {
    constructor() {
        this.color = 'dark';
        this.click = new EventEmitter();
    }
    ngOnInit() {
    }
    onClick(event) {
        if (this.disabled) {
            return;
        }
        this.click.emit(event);
        event.stopPropagation();
        event.preventDefault();
    }
}
CoolFacebookButtonComponent.decorators = [
    { type: Component, args: [{
                selector: 'cool-facebook-button',
                template: "<button type=\"button\" class=\"facebook-button\"\n        [class.disabled]=\"disabled\"\n        [class.light]=\"color === 'light'\"\n        [class.dark]=\"color === 'dark'\"\n        (click)=\"onClick($event)\">\n  <span class=\"icon\">\n    <svg width=\"20\" height=\"28\" viewBox=\"0 0 28 29\">\n      <path d=\"M9 8h-3v4h3v12h5v-12h3.642l.358-4h-4v-1.667c0-.955.192-1.333 1.115-1.333h2.885v-5h-3.808c-3.596 0-5.192 1.583-5.192 4.615v3.385z\"/>\n    </svg>\n  </span>\n  <span class=\"text\"><ng-content></ng-content></span>\n</button>",
                changeDetection: ChangeDetectionStrategy.OnPush,
                encapsulation: ViewEncapsulation.ShadowDom,
                styles: [".facebook-button{background-color:#3b5998;border:1px solid #3b5998;border-radius:5px;box-shadow:1px 1px 0 1px rgba(0,0,0,.05);color:#fff;cursor:pointer;height:36px;line-height:1;padding:0;transition-duration:.15s;transition-property:background-color,box-shadow;transition-timing-function:ease-in-out;white-space:nowrap}.facebook-button.dark{background-color:#3b5998;border:1px solid #3b5998;color:#fff}.facebook-button.light{background-color:#fff;border:1px solid #fff;color:#757575}.facebook-button:active,.facebook-button:focus,.facebook-button:hover{outline:0}.facebook-button:not(.disabled):active,.facebook-button:not(.disabled):focus,.facebook-button:not(.disabled):hover{box-shadow:1px 4px 5px 1px rgba(0,0,0,.1)}.facebook-button.disabled{background-color:rgba(0,0,0,.12);border-color:#eee;box-shadow:0 0 0 0 rgba(0,0,0,.2),0 0 0 0 rgba(0,0,0,.14),0 0 0 0 rgba(0,0,0,.12);color:rgba(0,0,0,.26);cursor:default}.facebook-button.disabled .icon{background-color:#eee}.facebook-button .icon{background-color:#fff;border-bottom-left-radius:5px;border-top-left-radius:5px;box-sizing:border-box;display:inline-block;height:100%;padding:4px 12px;vertical-align:middle;width:40px}.facebook-button .text{display:inline-block;font-size:14px;font-weight:500;padding:0 10px;vertical-align:middle}.facebook-button svg{fill:#3b5998}"]
            },] }
];
CoolFacebookButtonComponent.ctorParameters = () => [];
CoolFacebookButtonComponent.propDecorators = {
    disabled: [{ type: Input }],
    color: [{ type: Input }],
    click: [{ type: Output }]
};
